core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -13986,7 +13997,7 @@ public final User addUserWithID(String userName, Object credential, String[] rol
             }
 
             // Validate the password against provided regular expressions.
-            if (!checkUserPasswordValid(credentialObj)) {
+            if (!UserCoreUtil.getAskPasswordEnabledFromThreadLocal() && !checkUserPasswordValid(credentialObj)) {
```

Comment:
better to move this logic inside checkUserPasswordValid method

Reply:
`checkUserPasswordValid` method is used to validate credentials during the password update also. Is it ok to move this logic into that method?

Reply:
That should be fine. The expectation is to skip the validation if the thread-local is set.

Reply:
Fixed